src/components/Photo.jsx

```diff
@@ -1,15 +1,19 @@
 import styles from '../styles/photo'
 
 import React from 'react'
+import { Link } from 'react-router'
 
-const STACK_FILES_DOWNLOAD_PATH = 'http://cozy.local:8080/files/download'
+import { STACK_FILES_DOWNLOAD_PATH } from '../constants/config'
 
-const Photo = ({ photo }) => {
+export const Photo = ({ photo, router }) => {
```

Comment:
In fact react-router's `withRouter()` HOC injects more than just the router [as explained in this documentation](https://github.com/ReactTraining/react-router/blob/master/docs/API.md#injected-props).

So you could have:
```javascript
export const Photo = ({ photo, location }) => {
```

Besides, if you can decouple responsibilities by limiting dependencies, it will not be a problem for performance. So, injecting `router` object or `location` object or just a string will not be a matter for performances, imho.